libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -120,7 +120,7 @@ struct CopyTranslate: public yul::ASTCopier
 
 		return yul::Identifier{
 			_identifier.location,
-			yul::YulString{m_context.localVariable(*varDecl).name()}
+			yul::YulString{m_context.localVariable(*varDecl).commaSeparatedList()}
```

Comment:
This actually fixes an ICE that occurred in the test, since ``name()`` only works for types with one unnamed stack component, but memory references have one ``mpos`` slot. If you deem it necessary, I assert that it's only one slot here or have a special case for reference types instead.

Reply:
+1 for assertion